File: PocketBeagle-2/fade/python/main.py
import time
from beagle_linux_sdk import PwmLed
from beagle_linux_sdk.boards.pocketbeagle2 import P1_36

PERIOD = 255
DELAY = 0.05

led = PwmLed(P1_36)

while True:
    for i in range(5, PERIOD, 5):
        led.start(i, PERIOD)
        time.sleep(DELAY)

    for i in range(PERIOD, -1, -5):
        led.start(i, PERIOD)
        time.sleep(DELAY)

File: PocketBeagle-2/fade/rust/src/main.rs
use std::thread::sleep;
use std::time::Duration;

use beagle_helper::boards::pocketbeagle2::P1_36;
use beagle_helper::PwmLed;

fn main() {
    const PERIOD: usize = 255;
    const DELAY: Duration = Duration::from_millis(50);

    let led = PwmLed::new(P1_36).unwrap();

    loop {
        for i in (5..(PERIOD + 1)).step_by(5) {
            led.start(i, PERIOD).unwrap();
            sleep(DELAY);
        }

        for i in (0..(PERIOD - 4)).step_by(5).rev() {
            led.start(i, PERIOD).unwrap();
            sleep(DELAY);
        }
    }
}
